### PR TITLE
Allow ^2.0 and ^3.0 of psr/log

### PR DESCRIPTION
**Pull request by CiiDyR**
# Description
This PR seeks to add support for API-compatible versions of psr/log

## Issue reference
#126 

## Checklist
* [x] Tests pass
* [ ] Created/updated tests
* [ ] Extended the documentation

### CODE REVIEW

File: src/lib/PubSub/CloudEvent.php [hunk 1]
```diff
@@ -8,12 +8,14 @@
...
 use InvalidArgumentException;
 use JetBrains\PhpStorm\ArrayShape;
+use JetBrains\PhpStorm\Deprecated;
 use LogicException;
 
 /**
  * Class CloudEvent
  * @package Dapr\PubSub
  */
+#[Deprecated(since: '1.3.0', replacement: \CloudEvents\V1\CloudEvent::class)]
```
[the remark this answers is not in the document]
Comment (CiiDyR) [reply src/lib/PubSub/CloudEvent.php:18]: Definitely! I'll add that right away.
However I didn't really mean to push the CloudEvent replacement into the same PR as the psr/log updates :sweat_smile:

**Comment by CiiDyR** [comment]
@withinboredom The psr/log update should now be good to go. I'll send another PR for the CloudEvents replacement

**Comment by withinboredom** [comment]
I expect the integration tests to fail to build again, but while I fix them, and you're in the composer.json file, feel free to add yourself to the authors file :)

**Comment by CiiDyR** [comment]
Finally some public contributions besides fixing typos and updating docs :fire: :-)

**Comment by CiiDyR** [comment]
@withinboredom Same here, just added my signature so should be ready for merging.

**Pull request merged**